softlayer/resource_softlayer_virtual_guest.go

```diff
@@ -718,6 +765,29 @@ func resourceSoftLayerVirtualGuestRead(d *schema.ResourceData, meta interface{})
 	}
 	d.SetConnInfo(connInfo)
 
+	// Read secondary IP addresses
+	d.Set("secondary_ip_addresses", nil)
+	if result.PrimaryIpAddress != nil {
+		secondarySubnetResult, err := services.GetAccountService(meta.(ProviderConfig).SoftLayerSession()).
+			Mask("ipAddresses[id,ipAddress]").
+			Filter(filter.Build(filter.Path("publicSubnets.endPointIpAddress.ipAddress").Eq(*result.PrimaryIpAddress))).
+			GetPublicSubnets()
+		if err != nil {
+			log.Printf("Error getting secondary Ip addresses: %s", err)
+		}
+
+		secondaryIps := make([]string, 0)
```

Comment:
To get the number of secondary ips, I need to execute for loop and count it as follows:
```
		numberOfSecondaryIps := 0
		for _, subnet := range secondarySubnetResult {
			numberOfSecondaryIps += len(subnet)
		}
		secondaryIps := make([]string, 0, numberOfSecondaryIps)
```
It's difficult to expect optimization.